### DISCUSSION TITLE
Thumbnails

### DISCUSSION THREAD

**Discussion by DiHubKi** [MartinEesmaa/VVCEasy#20] [Ideas]
Is it possible to use this https://github.com/megakraken/FFmpegThumbnails (gives error on my pc)  or something to somehow implement vvc video thumbnails in windows explorer

**Comment by DiHubKi** [comment]
For now i do thumbnails with this https://www.bannerbear.com/blog/how-to-set-a-custom-thumbnail-for-a-video-file-using-ffmpeg/ it works but video is not showing anything anymore only sound  i dont know why (fixed with mp4box)

![image](https://user-images.githubusercontent.com/53359732/189512890-80bc3a8d-d1f8-47a8-ac67-6eca65e8429f.png)

![image](https://user-images.githubusercontent.com/53359732/189512953-734f35b7-48c7-4df7-a795-ad6942b16b29.png)

**Reply by MartinEesmaa** [reply]
Sorry, I fixed MPV.NET VVCEasy version. Link: https://github.com/MartinEesmaa/VVCEasy/commit/d68046070ba1df8e640b9313ce86583af7aa63c0

Also, you can safely remove libwinpthread-1.dll & libgcc_s_seh-1.dll.

- Martin Eesmaa

**Reply by DiHubKi** [reply]
For me it wasnt broken i just did it wrong but thank you

**Reply by DiHubKi** [reply]
If i use mp4box instead of ffmpeg to add thumbnail video is not broken anymore 
![image](https://user-images.githubusercontent.com/53359732/189517362-c7be84a4-6388-4456-8640-c9222d7f0050.png)

**Reply by MartinEesmaa** [reply]
> If i use mp4box instead of ffmpeg to add thumbnail video is not broken anymore

Yeah, but it does not show only VVC thumbnail, AV1 10-bit might not show the video thumbnail :/.

- Martin Eesmaa

**Comment by MartinEesmaa** [comment]
> Is it possible to use this https://github.com/megakraken/FFmpegThumbnails (gives error on my pc) or something to somehow implement vvc video thumbnails in windows explorer

@DisaKzionda, 
It requires to get lib, bin & include folders to show thumbnails on your PC: https://github.com/megakraken/FFmpegThumbnails/tree/main/FFmpegThumbnail/FFmpeg

I will make shared build with static of FFmpeg VVCEasy build folders (lib, include, lib), but it might be possible.

- Martin Eesmaa

**Reply by MartinEesmaa** [reply]
@DisaKzionda, I tried to build ffmpeg with shared dll by static, but it failed for only two undefined references on final linking:
```
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libavcodec/ansi.o:ansi.c:(.text+0x108e): undefined reference to `__imp_avpriv_cga_font'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libavcodec/ansi.o:ansi.c:(.text+0x1235): undefined reference to `__imp_avpriv_vga16_font'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libavcodec/ansi.o:ansi.c:(.text+0x1268): undefined reference to `__imp_avpriv_cga_font'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libavcodec/ansi.o:ansi.c:(.text+0x129b): undefined reference to `__imp_avpriv_cga_font'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libavcodec/ansi.o:ansi.c:(.text+0x12ce): undefined reference to `__imp_avpriv_cga_font'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libavcodec/ansi.o:ansi.c:(.text.unlikely+0x1c): undefined reference to `__imp_avpriv_vga16_font'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libavcodec/bintext.o:bintext.c:(.text.unlikely+0x12c): undefined reference to `__imp_avpriv_vga16_font'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libavcodec/bintext.o:bintext.c:(.text.unlikely+0x154): undefined reference to `__imp_avpriv_cga_font'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libavcodec/tmv.o:tmv.c:(.text+0xfa): undefined reference to `__imp_avpriv_cga_font'
collect2.exe: error: ld returned 1 exit status
make: *** [/home/marti/FFmpeg-FixVVC/ffbuild/library.mak:103: libavcodec/avcodec-58.dll] Error 1
```

I will try again next time tomorrow or other day.

- Martin Eesmaa

**Reply by DiHubKi** [reply]
I don't  understand how it all works but thanks for trying anyway.
